src/orange/BoundingBox.hh

```diff
@@ -110,8 +110,8 @@ CELER_FUNCTION BoundingBox<T>::BoundingBox(Real3 const& lo, Real3 const& hi)
  * Lower bbox coordinate (must be valid).
  */
 template<class T>
-CELER_FUNCTION typename BoundingBox<T>::Real3 const&
-BoundingBox<T>::lower() const
+CELER_FUNCTION auto BoundingBox<T>::lower() const ->
+    typename BoundingBox<T>::Real3 const&
```

Comment:
@elliottbiondo The nicest thing about trailing return types is that they act as if they're in class scope, so you literally only have to write `-> Real3`, thereby avoiding the `typename` etc. I thought the Real3 was part of the unintentional merge conflict resolution and you planned to revert it? Either way is fine.